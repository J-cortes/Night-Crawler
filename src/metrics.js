
function averageTime(collection) {
    return collection.reduce((sum, i) => sum + i.data.time, 0) / collection.length;
}
averageTime.displayName = 'Average Backend Time';

function percentSuccess(collection) {
    return collection.filter(i => i.data.status <= 500).length / collection.length * 100;
}
percentSuccess.displayName = '% Successful';

function totalRequests(collection) {
    return collection.length;
}
totalRequests.displayName = 'Total Requests';

module.exports = {
    averageTime,
    percentSuccess,
    totalRequests
}