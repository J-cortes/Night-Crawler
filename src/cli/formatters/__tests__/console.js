import ConsoleFormatter from '../console';
import Analysis from '../../../analysis';
import { Number } from '../../../metrics';

describe('Console Formatter', function() {
  it('Should output a listing of results marked according to level', function() {
    var a = new Analysis('test', new Date());
    a.addResult('OK', 0, 10, 'm1');
    a.addResult('WARN', 1, 10, 'm1');
    a.addResult('ERR', 2, 10, 'm1');
    expect(new ConsoleFormatter().format(a)).toMatchSnapshot();
  });
  it('Should output aggregates, marked by level', function() {
    var a = new Analysis('test', new Date());
    a.addMetric('ok', new Number('OK metric', 0, 0));
    a.addMetric('warn', new Number('WARN metric', 1, 2));
    a.addMetric('err', new Number('ERR metric', 2, 5));
    expect(new ConsoleFormatter().format(a)).toMatchSnapshot();
  });
  it('Should show no results when there are no results or metrics', function() {
    var a = new Analysis('test', new Date());
    expect(new ConsoleFormatter().format(a)).toMatchSnapshot();
  });
});
