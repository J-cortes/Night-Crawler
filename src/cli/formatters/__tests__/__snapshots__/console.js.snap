// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Console Formatter Should output a listing of results marked according to level 1`] = `
"Results
======
| Url  |
| ---- |
| OK   |
| [33mWARN[39m |
| [31mERR[39m  |

Metrics
=======
[33mNo Results[39m"
`;

exports[`Console Formatter Should output aggregates, marked by level 1`] = `
"Results
======
[33mNo Results[39m

Metrics
=======
| Name        | Value |
| ----------- | ----- |
| OK metric   | 0     |
| WARN metric | [33m2[39m     |
| ERR metric  | [31m5[39m     |"
`;

exports[`Console Formatter Should show no results when there are no results or metrics 1`] = `
"Results
======
[33mNo Results[39m

Metrics
=======
[33mNo Results[39m"
`;
